### ISSUE TITLE
Bug on GlobalSearch not expanding when interacted with the JAWS screen reader

### ISSUE THREAD

**Issue by aliunwala** [carbon-design-system/gatsby-theme-carbon#956]
This issue is also being tracked in our project here https://github.ibm.com/IBMa/E2E/issues/1254 
We see this issue on the https://ibm.com/able site as well as we are importing the Gatsby GlobalSearch implementation.

Title: Bug on GlobalSearch not expanding when interacted with the JAWS screen reader

### Describe in detail the issue you’re having.
When using JAWS, the search widget on the top bar won't open. It activates whatever the last focused item was. For example, if I open the switcher and then tab back to the search and hit enter, it opens the switcher and not the search. I am unable to reproduce this without JAWS running.

### What did you expect to happen? What happened instead? What would you like to see changed?
It should allow the user to search through keyboard navigation.

### What browser are you working in?
Browser-agnostic, happens with JAWS running on top of any browser

## Steps to reproduce the issue
1. Step one: Open JAWS
2. Step two: Navigate to the GlobalSearch field by use of the tab key 
3. Step three: Press enter to expand the search (This currently does not work)

## Fix
I have a PR that I will submit with this issue that should help with a possible resolution for this bug. The fix we found for this was the remove the role="combobox" entirely from the code here (Line 159):
https://github.com/carbon-design-system/gatsby-theme-carbon/blob/main/packages/gatsby-theme-carbon/src/components/GlobalSearch/GlobalSearch.js#L159

This works as the combobox does not describe the field well and so JAWS thinks this section of the page is not expandable and does not allow the enter key on that div: 

Current code:
![image](https://user-images.githubusercontent.com/4451128/89254174-25b37880-d5e4-11ea-9b02-9c03ebe5e4c5.png)

Fixed code:
![image](https://user-images.githubusercontent.com/4451128/89254472-cc981480-d5e4-11ea-9b77-3de6af531d5e.png)

I also was able to run the "example" site on localhost:8000 and did not see this one line change break any page functionality. 

Thanks!

- Screenshots or code
Here is the location I am talking about:
![image](https://user-images.githubusercontent.com/4451128/89253816-462f0300-d5e3-11ea-9d95-f9591fd273a4.png)

- Notes
None

**Comment by aliunwala** [comment]
PR opened here:
https://github.com/carbon-design-system/gatsby-theme-carbon/pull/957
